backend/tests/Api/SubmissionTest.php

```diff
@@ -649,4 +649,207 @@ public function testUpdatedByFieldIsSet()
         $response->assertJsonPath('data.updateSubmission.created_by.id', (string)$submission->created_by);
         $response->assertJsonPath('data.updateSubmission.updated_by.id', (string)$user->id);
     }
+
+    /**
+     * @param Submission $submission
+     * @param string $status
+     * @return void
+     */
+    protected function executeSubmissionStatusUpdateSuccessfully(Submission $submission, string $status)
+    {
+        $response = $this->graphQL(
+            'mutation UpdateSubmissionStatus ($submission_id: ID!, $status: SubmissionStatus) {
+                updateSubmission(
+                    input: {
+                        id: $submission_id
+                        status: $status
+                    }
+                ) {
+                    id
+                    status
+                }
+            }',
+            [
+                'submission_id' => $submission->id,
+                'status' => $status,
+            ]
+        );
+        $expected_data = [
+            'updateSubmission' => [
+                'id' => (string)$submission->id,
+                'status' => $status,
+            ],
+        ];
+        $response->assertJsonPath('data', $expected_data);
+    }
+
+    /**
+     * @param Submission $submission
+     * @param string $status
+     * @return void
+     */
+    protected function executeSubmissionStatusUpdateUnsuccessfully(Submission $submission, string $status)
+    {
+        $response = $this->graphQL(
+            'mutation UpdateSubmissionStatus ($submission_id: ID!, $status: SubmissionStatus) {
+                updateSubmission(
+                    input: {
+                        id: $submission_id
+                        status: $status
+                    }
+                ) {
+                    id
+                    status
+                }
+            }',
+            [
+                'submission_id' => $submission->id,
+                'status' => $status,
+            ]
+        );
+        $response->assertJsonPath('errors.0.extensions.category', 'authorization');
+    }
+
+    /**
+     * @return array
+     */
+    public function allSubmissionStates(): array
```

Comment:
```suggestion
    public function provideAllSubmissionStates(): array
```

I'm confident there's other places we've not followed a convention on this, but we should try to follow a provide* convention for dataproviders.

Reply:
Refactored as suggested